index.ts

```diff
@@ -0,0 +1,154 @@
+// Neara Infrastructure Task
+
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+const stack = pulumi.getStack();
+const tags = { Name: "neara-task" };
+
+// Create VPC and subnets
+const vpc = new aws.ec2.Vpc("andy-vpc", {
+  cidrBlock: "10.0.0.0/16",
+  tags: tags,
+});
+
+const privateSubnet = new aws.ec2.Subnet("private-subnet", {
+  vpcId: vpc.id,
+  cidrBlock: "10.0.1.0/24",
+  tags: tags,
+});
+
+const publicSubnet = new aws.ec2.Subnet("public-subnet", {
+  vpcId: vpc.id,
+  cidrBlock: "10.0.2.0/24",
+  tags: tags,
+});
+
+// Lambda
+const lambdaRole = new aws.iam.Role("info-handler-role", {
+  assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal(
+    aws.iam.Principals.LambdaPrincipal
+  ),
+  managedPolicyArns: [
+    aws.iam.ManagedPolicy.AWSLambdaBasicExecutionRole,
+    aws.iam.ManagedPolicy.AWSLambdaVPCAccessExecutionRole,
+  ],
+  tags: tags,
+});
+
+const infoHandler = new aws.lambda.Function("info-handler", {
+  role: lambdaRole.arn,
+  runtime: aws.lambda.Runtime.Python3d9,
+  code: new pulumi.asset.FileArchive("./lambda/main.py.zip"),
+  handler: "main.info_handler", // fileName.methodName
+  vpcConfig: {
+    subnetIds: [privateSubnet.id],
+    securityGroupIds: [vpc.defaultSecurityGroupId],
+  },
+  tags: tags,
+});
+
+// API Gateway
+const securityGroup = new aws.ec2.SecurityGroup("api-gateway-security-group", {
+  vpcId: vpc.id,
+  ingress: [
+    {
+      description: "HTTP inbound",
+      fromPort: 80,
+      toPort: 80,
+      protocol: "tcp",
+      cidrBlocks: ["0.0.0.0/0"],
+    },
+  ],
+  tags: tags,
+});
+
+const vpcLink = new aws.apigatewayv2.VpcLink("andy-vpc-link", {
+  subnetIds: [privateSubnet.id],
+  securityGroupIds: [securityGroup.id],
+  tags: tags,
+});
+
+const api = new aws.apigatewayv2.Api("andy-api-gateway", {
```

Comment:
Curiousity: why do you have `andy` on the `resourceName`s?

Reply:
Inconsistency on my part